Code/GraphMol/Depictor/Templates.h

```diff
@@ -0,0 +1,72 @@
+//
+//  Copyright (C) 2003-2022 Greg Landrum and other RDKit contributors
+//
+//   @@ All Rights Reserved @@
+//  This file is part of the RDKit.
+//  The contents are covered by the terms of the BSD license
+//  which is included in the file license.txt, found at the root
+//  of the RDKit source tree.
+//
+
+
+#include <GraphMol/ROMol.h>
+#include <GraphMol/SmilesParse/SmilesParse.h>
+#include <GraphMol/MolOps.h>
+
+#include <iostream>
+#include <fstream>
+#include <unordered_map>
+
+class CoordinateTemplates {
+ public:
+  //! returns a reference to the singleton CoordinateTemplates
+  /*
+      \return a reference to the singleton CoordinateTemplates
+
+      <b>Notes:</b>
+        - if the singleton CoordinateTemplates has already been instantiated
+          the singleton will be returned, otherwise the singleton will
+          be constructed.
+
+   */
+  static CoordinateTemplates& getRingSystemTemplates() {
+    static CoordinateTemplates template_mols;
+    return template_mols;
+  }
+
+  bool hasTemplateOfSize(unsigned int atom_count) {
+    if (m_templates.find(atom_count) != m_templates.end()) {
+        return true;
+    }
+    return false;
+  }
+
+  const std::vector<std::shared_ptr<RDKit::ROMol>>& getMatchingTemplates(unsigned int atom_count) {
+    return m_templates[atom_count];
+  }
+
+ private:
+  CoordinateTemplates() {
+    // load templates into m_templates map by atom count
+    std::string rdbase = getenv("RDBASE");
```

Comment:
Are you suggesting we store all templates in a single string or vector within this header, or create another header for them? I am happy to do that as long as it would still be an OK format if we had far more templates (we have 70 right now, but I could see having a couple hundred).

I've been hesitant about allowing users to submit their own templates, since they must adhere to specific requirements such as being ring systems and having standard bond lengths. Users could achieve their desired structure with alignment code, and they could add or replace templates through pull requests.

That being said, you're probably right, it would be useful (especially for organizations) to be able to define their own templates. What do you think about making it a cmake option, that way we users can't change the templates around at runtime? I'm open to other ideas as well.

Reply:
I think it's a `std::vector<string>` and can be in a `.h` or `.in` file.

Reply:
Ok, I added this in a new header -- lmk what you think